src/engine/pieces/piece.ts

```diff
@@ -13,6 +14,35 @@ export default class Piece {
         throw new Error('This method must be implemented, and return a list of available moves');
     }
 
+    public getLateralSquares(board: Board) {
+        let currentSquare: Square = board.findPiece(this);
+        let lateralSquares: Square[] = [];
+
+        for(let i: number = 0; i < GameSettings.BOARD_SIZE; i++) {
+            if (i !== currentSquare.col) lateralSquares.push(new Square(currentSquare.row, i));
+            if (i !== currentSquare.row) lateralSquares.push(new Square(i, currentSquare.col));
+        }
+
+        return lateralSquares;
+    }
+
+    public getDiagonalSquares(board: Board) {
+        let currentSquare: Square = board.findPiece(this);
+        let diagonalSquares: Square[] = [];
+        let currentRow = currentSquare.row;
+        let currentCol = currentSquare.col;
+        let displacement;
+
+        for (let i = 0; i <8; i++){
```

Comment:
As per the previous PR, this 8 is a bit suspect.

What we mean here is the board size as you've used above, but that's then implicitly a property of the board.